src/main/scala/io/iohk/ethereum/App.scala

```diff
@@ -1,59 +1,101 @@
   val blockchain: Blockchain = BlockchainImpl(storagesInstance.storages)
 
   def main(args: Array[String]): Unit = {
     val actorSystem = ActorSystem("etc-client_system")
 
-    val nodeStatus =
-      NodeStatus(
-        key = nodeKey,
-        serverStatus = ServerStatus.NotListening,
-        blockchainStatus = BlockchainStatus(Config.Blockchain.genesisDifficulty, Config.Blockchain.genesisHash, 0))
+    val appActor = actorSystem.actorOf(AppActor.props(nodeKey, blockchain, storagesInstance, actorSystem), "app")
+
+    appActor ! AppActor.StartApp
+  }
+
+}
+
+class AppActor(nodeKey: AsymmetricCipherKeyPair,
+               blockchain: Blockchain,
+               storagesInstance: DataSourcesComponent with StoragesComponent,
+               actorSystem: ActorSystem) extends Actor {
+
+  import Config.{Network => NetworkConfig}
+  import AppActor._
+
+  override def receive: Receive = idle
 
-    val nodeStatusHolder = Agent(nodeStatus)
+  def idle: Receive = {
 
-    val peerManager = actorSystem.actorOf(PeerManagerActor.props(nodeStatusHolder, Config.Network.peer, blockchain), "peer-manager")
-    val server = actorSystem.actorOf(ServerActor.props(nodeStatusHolder, peerManager), "server")
+    case StartApp =>
+      val nodeStatus =
+        NodeStatus(
+          key = nodeKey,
+          serverStatus = ServerStatus.NotListening,
+          blockchainStatus = BlockchainStatus(Config.Blockchain.genesisDifficulty, Config.Blockchain.genesisHash, 0))
 
-    server ! ServerActor.StartServer(NetworkConfig.Server.listenAddress)
+      val nodeStatusHolder = Agent(nodeStatus)
 
-    if(Config.Network.Rpc.enabled) JsonRpcServer.run(actorSystem, blockchain, Config.Network.Rpc)
+      val peerManager = actorSystem.actorOf(PeerManagerActor.props(nodeStatusHolder, Config.Network.peer, blockchain), "peer-manager")
+      val server = actorSystem.actorOf(ServerActor.props(nodeStatusHolder, peerManager), "server")
 
-    val fastSyncController = actorSystem.actorOf(
-      FastSyncController.props(
-        peerManager,
-        nodeStatusHolder,
-        blockchain,
-        storagesInstance.storages.mptNodeStorage),
-      "fast-sync-controller")
+      server ! ServerActor.StartServer(NetworkConfig.Server.listenAddress)
 
-    fastSyncController ! FastSyncController.StartFastSync
+      if(Config.Network.Rpc.enabled) JsonRpcServer.run(actorSystem, blockchain, Config.Network.Rpc)
 
-    Runtime.getRuntime.addShutdownHook(new Thread() {
-      override def run(): Unit = {
-        storagesInstance.dataSources.closeAll
+      val fastSyncController = actorSystem.actorOf(
+        FastSyncController.props(
+          peerManager,
+          nodeStatusHolder,
+          blockchain,
+          storagesInstance.storages.mptNodeStorage),
+        "fast-sync-controller")
+
+      fastSyncController ! FastSyncController.StartFastSync
+
+      Runtime.getRuntime.addShutdownHook(new Thread() {
```

Comment:
Not if we move `storagesInstance` inside the `AppActor`. As we don't use `storagesInstance` outside of `AppActor` what do you think of keeping it here and creating the `storagesInstance` here as well?